app/services/react_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class ReactService < BaseService
+  include Authorization
+  include Payloadable
+
+  def call(account, status, emoji)
+    authorize_with account, status, :react?
+
+    name, domain = emoji.split('@')
+    return unless domain.nil? || status.local?
+
+    normalized = "#{name}\uFE0F"
+    name = normalized if StatusReactionValidator::SUPPORTED_EMOJIS.include?(normalized)
```

Comment:
The same needs to be applied to the unreact service, and it might break if a server were to send a normalised and unnormalised emoji.

Maybe it's best to move this transform to the API interface?

Reply:
Thanks for the catch, moved it.

Reply:
Not quite yet; the fix ([`a82d4e5` (#2462)](https://github.com/glitch-soc/mastodon/pull/2462/commits/a82d4e58eb931159eb3c17573b63a2c8b41f3871)) moves it to catch the `#create` and `#destroy` urls, but the problem was in the display.

I meant to also find a way to merge non-normalised and normalised reacts when requesting reacts per status, and/or also per the notification service, if that's not a huge deal.

The goal is also to prevent duplicate reacts (that a non-normalised and a normalised emoji become 2 of normalised) and etc, but i dont know how much of an effort that is.

Reply:
just spitballing because i've taken like the most cursory glance at this particular issue and how it's implemented, but can we handle / store emoji objects as potentially containing both a normalized and a non-normalized emoji? and then choose one to prefer to display? I think that would solve the issue how merging theme and preventing duplicate reacts whilst also not messing up create / destroy actions